### ISSUE TITLE
new feature: methods to add new vocab/wordunits while fine-tuning

### ISSUE THREAD

**Issue by neuspell** [neuspell/neuspell#1]
Models trained for spell-correction can be fine-tuned for use-case specific dataset, which belongs to a textual domain with vocabulary different from the pre-trained dataset. Thus, at times, it is required to train spell correction models to predict words that are currently marked as OUT-OF-VOCABULARY. For example, the word _lexical_ is not available in the currently used 100K word-list (see scripts/seq_modeling/wordunits). Furthermore, once some new vocabulary is added, the output layer of the model architecture has to be changed. This also means that the final layer has to be trained again in the use-case specific training.

**Comment by Atul997** [comment]
Thanks for open-sourcing this wonderful library. I am waiting for the feature to add vocabulary or words at the runtime.

Thanks

**Comment by murali1996** [comment]
Interested practitioners can follow the code snippet from [here](https://github.com/neuspell/neuspell/blob/227c6e7b2db0c38f18e6feb9d8004ecfa01ae233/neuspell/corrector_subwordbert.py#L121) to provide this support. We will soon look into providing this support as we look forward to any open source contributions.

**Comment by dhhcj1** [comment]
wonderful work , and i would like to confirm whether the project can implement the fine-tuning work at present ?  really looking forward you reply, thank you

**Comment by murali1996** [comment]
Yes, fine-tuning is now possible for two of the best performing models. See the issue #12 for more details.

**Comment by vjkadekar** [comment]
thanks for work murali... but why torch 1.4? Cant we use 1.7?

**Comment by murali1996** [comment]
Hi @vjkadekar, the dependency is extended to the latest versions of torch and transformers. You can also do `pip install neuspell` now.
